From 874e9a963c55bc06942fc84b35dbc3da02075308 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Thu, 23 Aug 2012 15:52:17 +0530
Subject: [PORT FROM MAIN] Jellybean: Enabling AAC-ELD decoding support.

BZ: 49870

Patch - 2
Currently MDP Intel optimized AAC decoder suite doesn't have AAC-ELD
decoding capability.
This fix provides capability to invoke Google-AAC-ELD decoding ONLY
for AAC-ELD profiles and invoke the default Intel MDP optimized AAC
decoding for other profiles.

Change-Id: I021b8f1b2417e769b7cd28c0a91b1348f7644123
Orig-Change-Id: I0276277e500c61535a363e82a10e654413a5ca9f
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
Reviewed-on: http://android.intel.com:8080/77919
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Kandasamy, Muthukumar <muthukumar.kandasamy@intel.com>
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 include/media/stagefright/MediaDefs.h   |    4 +++-
 media/libstagefright/MPEG4Extractor.cpp |    4 ++++
 media/libstagefright/MediaDefs.cpp      |    4 +++-
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/include/media/stagefright/MediaDefs.h b/include/media/stagefright/MediaDefs.h
index 81de6e4..b105fdd 100644
--- a/include/media/stagefright/MediaDefs.h
+++ b/include/media/stagefright/MediaDefs.h
@@ -56,7 +56,9 @@ extern const char *MEDIA_MIMETYPE_CONTAINER_WVM;
 
 extern const char *MEDIA_MIMETYPE_TEXT_3GPP;
 extern const char *MEDIA_MIMETYPE_TEXT_SUBRIP;
-
+#ifdef USE_INTEL_MDP
+extern const char *MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED;
+#endif
 }  // namespace android
 
 #endif  // MEDIA_DEFS_H_
diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 3503aaf..e789230 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -2199,7 +2199,11 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
 
     if (objectType == 31) {  // AAC-ELD => additional 6 bits
         objectType = 32 + br.getBits(6);
+#ifdef USE_INTEL_MDP
+        mLastTrack->meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED);
+#endif
     }
+    mLastTrack->meta->setInt32(kKeyAACProfile, objectType);
 
     uint32_t freqIndex = br.getBits(4);
 
diff --git a/media/libstagefright/MediaDefs.cpp b/media/libstagefright/MediaDefs.cpp
index 5d8029c..eaa20dc 100644
--- a/media/libstagefright/MediaDefs.cpp
+++ b/media/libstagefright/MediaDefs.cpp
@@ -54,5 +54,7 @@ const char *MEDIA_MIMETYPE_CONTAINER_WVM = "video/wvm";
 
 const char *MEDIA_MIMETYPE_TEXT_3GPP = "text/3gpp-tt";
 const char *MEDIA_MIMETYPE_TEXT_SUBRIP = "application/x-subrip";
-
+#ifdef USE_INTEL_MDP
+const char *MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED = "audio/mp4a-aacextended"; //For Audio Object Type > 31
+#endif
 }  // namespace android
-- 
1.7.4.1

